### PR TITLE
More LuaLS annotations

### PR DESCRIPTION
**Pull request by emmericp**
Down to 0* errors :)

*together with some definition files that I haven't yet published

### CODE REVIEW

**Comment by emmericp** [comment]
WIP: need to clean up the vscode stuff, i want that in another PR

File: DBM-GUI/DBM-GUI.lua [hunk 1]
```diff
@@ -518,7 +513,7 @@ function DBM_GUI:CreateBossModPanel(mod)
 					else
 						local _title = DBM:GetSpellInfo(spellID)
 						if _title then
-							title, desc, icon = _title, tonumber(spellID), GetSpellTexture(spellID)
+							title, desc, icon = _title, tonumber(spellID), GetSpellTexture(spellID or 0)
```
Comment (QartemisT) [inline DBM-GUI/DBM-GUI.lua:516]: What's going on with this line?

File: DBM-GUI/modules/MainFramePrototype.lua [hunk 2]
```diff
@@ -251,6 +254,7 @@ function frame:CreateTab(tab)
 	self.tabs[i] = tab
 	DBM_GUI:CreateNewFauxScrollFrameList()
 	---@class DBMOptionsFrameTabButtonTemplate: Button
+	---@field Text FontString -- (no clue where this field is even from?)
```
Comment (QartemisT) [inline DBM-GUI/modules/MainFramePrototype.lua:257]: The field is gained from the template mixin.

File: DBM-Core/modules/TargetScanning.lua [hunk 3]
```diff
@@ -140,7 +141,7 @@ do
 		--Perform normal scan criteria matching
 		elseif targetname and targetname ~= CL.UNKNOWN and (not targetFilter or (targetFilter and targetFilter ~= targetname)) then
 			if not IsInGroup() then scanTimes = 1 end--Solo, no reason to keep scanning, give faster warning. But only if first scan is actually a valid target, which is why i have this check HERE
-			if (isEnemyScan and UnitIsFriend("player", targetuid) or (onlyPlayers and not UnitIsPlayer("player", targetuid)) or mod:IsTanking(targetuid, bossuid)) and not isFinalScan then--On player scan, ignore tanks. On enemy scan, ignore friendly player. On Only player, ignore npcs and pets
+			if (isEnemyScan and UnitIsFriend("player", tostring(targetuid)) or (onlyPlayers and not UnitIsPlayer(targetuid)) or mod:IsTanking(targetuid, bossuid)) and not isFinalScan then--On player scan, ignore tanks. On enemy scan, ignore friendly player. On Only player, ignore npcs and pets
```
Comment (QartemisT) [inline DBM-Core/modules/TargetScanning.lua:144]: What's the purpose of tostring here? The only occurances of targetuid is a string (check checking inside of the function returns)

File: DBM-Core/modules/Sounds.lua [hunk 4]
```diff
@@ -105,6 +110,7 @@ do
 
 	local hasCached = false
 	local cachedTable
+	---@deprecated Use new utility functions
```
Comment (QartemisT) [inline DBM-Core/modules/Sounds.lua:113]: Should we remove the other deprecated notice on the line below then?

File: DBM-Core/modules/Sounds.lua [hunk 5]
```diff
@@ -151,6 +157,7 @@ do
...
```
Comment (QartemisT) [inline DBM-Core/modules/Sounds.lua:160]: Should we remove the other deprecated notice on the line below then?

File: DBM-Core/modules/Sounds.lua [hunk 6]
```diff
@@ -48,6 +52,7 @@ do
 	}
 	local hasCached = false
 	local cachedTable
+	---@deprecated Use new utility functions
```
Comment (QartemisT) [inline DBM-Core/modules/Sounds.lua:55]: Should we remove the other deprecated notice on the line below then?

File: DBM-Core/modules/Sounds.lua [hunk 7]
```diff
@@ -19,7 +22,8 @@ do
 	}
 	local hasCached = false
 	local cachedTable
-	DBM.Counts = counts -- @Deprecated: Use new utility functions
+	---@deprecated Use new utility functions
```
Comment (QartemisT) [inline DBM-Core/modules/Sounds.lua:25]: Should we remove the other deprecated notice on the line below then?

File: DBM-Core/DBM-RangeCheck.lua [hunk 8]
```diff
@@ -654,7 +663,7 @@ do
 
 	function updateIcon()
 		local numPlayers = GetNumGroupMembers()
-		activeDots = max(numPlayers, activeDots)
+		activeDots = max(numPlayers or 0, activeDots)
```
Comment (QartemisT) [inline DBM-Core/DBM-RangeCheck.lua:666]: GetNumGroupMembers always returns a number? I see no documented occurances of it being nilable.

File: DBM-GUI/DBM-GUI.lua (hunk 1 above)
Comment (emmericp) [reply DBM-GUI/DBM-GUI.lua:516]: LuaLS isn't smart enough to deduce that spellID is guaranteed non-nil here, so the "or 0" is the quickest way to make it happy

File: DBM-GUI/modules/MainFramePrototype.lua (hunk 2 above)
Comment (emmericp) [reply DBM-GUI/modules/MainFramePrototype.lua:257]: Ah, never seen that one before, I'll fix that upstream

File: DBM-Core/modules/TargetScanning.lua (hunk 3 above)
Comment (emmericp) [reply DBM-Core/modules/TargetScanning.lua:144]: the relationship non-nil relationship between the return parameters of GetBossTarget above can't be encoded in the type system, so it doesn't know that targetname non nil implies that targetuid is also non-nil. i can change to check for targetuid in the if instead

File: DBM-Core/DBM-RangeCheck.lua (hunk 8 above)
Comment (emmericp) [reply DBM-Core/DBM-RangeCheck.lua:666]: unfortunately the Blizzard docs are missing this function and quite some other functions :(

I need to figure out how the type annotations on the wiki work to fix this upstream, see https://github.com/Ketho/vscode-wow-api/issues/124

File: DBM-Core/modules/Sounds.lua (hunk 4 above)
Comment (emmericp) [reply DBM-Core/modules/Sounds.lua:113]: done

**Review by QartemisT — APPROVED** [review]

**Pull request merged**